### PR TITLE
Escape $ in markdown html output

### PR DESCRIPTION
**Pull request by dunhamsteve**
Pandoc is failing to process @ohad 's setoid paper because `$` is not being escaped in markdown html output.  This patch adds `$` to the list of characters being escaped in markdown html.

### CODE REVIEW

File: src/Katla/Markdown.idr [hunk 1]
```diff
@@ -15,6 +15,8 @@ escapeMarkdown : Config -> Char -> List Char
 escapeMarkdown config ' ' = unpack config.space
 escapeMarkdown config '_' = unpack "\\_"
 escapeMarkdown config '*' = unpack "\\*"
+escapeMarkdown config '$' = unpack "\\$"
+escapeMarkdown config '\\' = unpack "\\\\"
```
Comment (ohad) [inline src/Katla/Markdown.idr:19]: Another option is:
```idris
escapeMarkdown config '\\' = unpack "&#92;"
```
I don't know what would be better. @gallais ?

Comment (ohad) [reply src/Katla/Markdown.idr:19]: There should be a similar fix `Katla.HTML`

**Comment by ohad** [comment]
Add tests, for example, in `tests/examples/markdown/Source.md`:
```idris
main = putStrLn
     $ unwords
     [ "Hello,"
     , (\x => x) "from"
     , "the"
     , "Markdown"
     , "mode"
     ]
```
and similarly in `tests/examples/standalone/Rainbow.idr`:
```idris
Rainbow n =  (\x => x) [ n , m , n ]
```
(and similarly add tests for `$`)

**Review by ohad — CHANGES_REQUESTED** [review]

File: src/Katla/Markdown.idr (hunk 1 above)
Comment (dunhamsteve) [reply src/Katla/Markdown.idr:19]: I don't understand the comment about Katla.HTML - I think the `htmlEscape` in there covers all of the cases that are necessary for HTML. The characters that could cause trouble there are `&` and `<`.

I've updated the markdown test to account for the `$` having a `\` in front of it and added a `$ (\x => x)` line to test the escaping of the backslash. 

@gallais - Let me know if we want to change any or all of these to numeric html entities. I don't have a strong opinion either way.  The `\\` is fine everywhere, and `\$` is fine with the modern processors that I've tested (pandoc, github, cmark), and it's officially in the commonmark standard). But `\$` does not work with the original gruber markdown script - it renders as-is.

Comment (gallais) [reply src/Katla/Markdown.idr:19]: > But \$ does not work with the original gruber markdown script - it renders as-is.

I don't know what the gruber is but [on the idris2 CI we are using `markdown`](https://github.com/idris-lang/Idris2/blob/main/www/katla.sh#L14) so the proposed change should either
1. work with it
2. start with a patch of the idris2 process to use something that will be compatible with the proposed change

Comment (dunhamsteve) [reply src/Katla/Markdown.idr:19]: I was a little terse there.  John Gruber was the original inventor of markdown and published the original perl script that processed markdown.

There are a number of packages that install an executable named `markdown` (markdown, multimarkdown, discount, etc), but I took a look at the Idris2 CI, and it is installing the `markdown` Ubuntu package which appears to be the original script by John Gruber. 

So we'll need to use the entity for `$` to be compatible with both that markdown and pandoc. That's either `&#36;`, `&#x24;`, or  `&dollar;`. If we end up taking option 2 for other reasons (like equation support), I think this will still net us wider compatibility.

Comment (ohad) [reply src/Katla/Markdown.idr:19]: > I don't understand the comment about Katla.HTML - I think the htmlEscape in there covers all of the cases that are necessary for HTML. The characters that could cause trouble there are `&` and `<`.

So long as we have tests that cover the edge cases in both the HTML backend (from `standalone`, say) and markdown (so not just the `\`, but also the other escape characters), then that's fine. 

Re. escaping dollar to be compatible with Gruber's script: since the output markdown isn't really human consumable anyway, I'd opt for whichever solution works on more backends.

Comment (ohad) [reply src/Katla/Markdown.idr:19]: I've added a test for the html backend:
https://github.com/dunhamsteve/katla/pull/1
if OK to merge into this PR, I'm happy we merge this PR

**Review by ohad — APPROVED** [review]
